local_pathfinding/objectives.py

```diff
@@ -40,19 +41,64 @@ class Objective(ob.StateCostIntegralObjective):
     - This class inherits from the OMPL class StateCostIntegralObjective:
         https://ompl.kavrakilab.org/classompl_1_1base_1_1StateCostIntegralObjective.html
     - Camelcase is used for functions that override OMPL functions, as that is their convention.
+    - Also computes the maximum motion cost for normalization purposes.
 
     Attributes:
         space_information (StateSpacePtr): Contains all the information about
             the space planning is done in.
+        max_motion_cost (float): The maximum motion cost between any two states in the state space.
     """
 
-    def __init__(self, space_information):
+    def __init__(self, space_information, num_samples: int):
         super().__init__(si=space_information, enableMotionCostInterpolation=True)
         self.space_information = space_information
 
+        states = self.sample_states(num_samples)
+        self.max_motion_cost = 1.0
+        self.max_motion_cost = self.find_maximum_motion_cost(states)
```

Comment:
I forgot, is there a reason why you initialize it to 1 first? Would it be possible to do this inside `find_maximum_motion_cost()`?

Reply:
Yes, we divide the motionCost() by max_motion_cost. So initially when we don't know the max_motion_cost we divide it as 1. Then sample and find the max_motion_cost. Finally normalise using the new max_motion_cost we obtain from the sample

Reply:
Added a comment to explain this